migration-engine/connectors/sql-migration-connector/src/sql_destructive_change_checker.rs

```diff
@@ -176,6 +176,7 @@ impl SqlMigrationConnector {
                             let columns = tables.columns(column_ides);
 
                             let arity_change_is_safe = match (&columns.previous().arity(), &columns.next().arity()) {
+                                (a, b) if a == b => true,
```

Comment:
I don't understand this, and I don't understand why there is a mention to SQLite a few lines below. If we want to harmonize things, I'd rather we add `(ColumnArity::List, ColumnArity::List)` to the second to last branch and remove the last one.